File: src/main/java/ConvertBSTToGreaterTree/BIGBOSS/ConvertBSTToGreaterTree.cpp
//
// Created by BIGBOSS on 2019-06-27.
//

/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution
{
    public:

    int sum = 0;

    TreeNode* convertBST(TreeNode* root)
    {
        if (root)
        {
            convertBST(root->right);
            sum += root->val;
            root->val = sum;
            convertBST(root->left);
        }

        return root;
    }
};

File: src/main/java/ConvertBSTToGreaterTree/zhangwentao/ConvertBSTToGreaterTree.py
#!/usr/bin/python
# -*- coding: UTF-8 -*-
#!@Author:zhangwentao

# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def convertBST(self, root):
        """
        :type root: TreeNode
        :rtype: TreeNode
        """
        lists = self.convertBSTExe(root)
        for i in range(1, len(lists), 1):
            lists[i].val += lists[i - 1].val
        return root
    def convertBSTToList(self,root):
        list = []
        if root == None: return list
        listOfRight = self.convertBSTToList(root.right)
        listOfLeft = self.convertBSTToList(root.left)
        list.extend(listOfRight)
        list.append(root)
        list.extend(listOfLeft)
        return list

